src/opaque.rs

```diff
@@ -0,0 +1,236 @@
+use std::{
+    io::{Cursor, Write},
+    marker::PhantomData,
+};
+
+use byteorder::{BigEndian, ReadBytesExt, WriteBytesExt};
+
+use crate::Error;
+
+// Opaque is a Variable-length Array that holds an uninterpreted byte array
+//https://datatracker.ietf.org/doc/html/rfc1014#section-3.12
+#[derive(Debug, Clone, Copy, PartialEq)]
+pub struct Opaque<T>
```

Comment:
After a bit of compiler fighting, we can make this:

```suggestion
pub(crate) struct Opaque<T>
```

And then we'll know it's not in the public API 👍